src/pages/RedirectFromOAuth.js

```diff
@@ -10,7 +10,13 @@ const RedirectFromOAuth = () => {
 
   const url = window.location.href;
   const isSilent = window.location.pathname.startsWith('/redirect-from-oauth-silent');
-  useOnMount(() => (isSilent === true ? userManager.signinSilentCallback(url) : userManager.signinPopupCallback(url)));
+  useOnMount(() => {
+    if (isSilent === true) {
```

Comment:
Are ternaries not allowed in React 18? Or is it that it is evaluating the function eagerly in the ternary now?

Reply:
The ternary is fine. The potential issue is returning the result of these functions from the effect function.

Reply:
I'm confused though because they should return nothing or a cleanup function. Aren't these returning the functions we want to execute on mount?

Reply:
These two user manager methods return Promises, not cleanup functions.
https://authts.github.io/oidc-client-ts/classes/UserManager.html#signinSilentCallback
https://authts.github.io/oidc-client-ts/classes/UserManager.html#signinPopupCallback

Reply:
Although you could've kept the ternary and added `{}`s to fix this.
```
  useOnMount(() => {
      isSilent === true ? userManager.signinSilentCallback(url) : userManager.signinPopupCallback(url);
  });
```

although of course, why not
```
  useOnMount(() => {
      (isSilent === true ? userManager.signinSilentCallback : userManager.signinPopupCallback)(url);
  });
```

Reply:
The if/else style seems clearer and easier to read to me, especially compared to the FP style example.